bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentBinarySensor.java

```diff
@@ -35,23 +41,41 @@
 
         protected @Nullable String device_class;
         protected boolean force_update = false;
-        protected int expire_after = 0;
+        protected @Nullable Integer expire_after;
+        protected @Nullable Integer off_delay;
 
         protected String state_topic = "";
         protected String payload_on = "ON";
         protected String payload_off = "OFF";
+
+        protected @Nullable String json_attributes_topic;
+        protected @Nullable String json_attributes_template;
+        protected @Nullable List<String> json_attributes;
     }
 
     public ComponentBinarySensor(CFactory.ComponentConfiguration componentConfiguration) {
         super(componentConfiguration, ChannelConfiguration.class);
 
-        if (channelConfiguration.force_update) {
-            throw new UnsupportedOperationException("Component:Sensor does not support forced updates");
+        OnOffValue value = new OnOffValue(channelConfiguration.payload_on, channelConfiguration.payload_off);
+
+        buildChannel(sensorChannelID, value, "value", getListener(componentConfiguration, value))//
+                .stateTopic(channelConfiguration.state_topic, channelConfiguration.value_template)//
```

Comment:
What's the purpose of these comments?

Reply:
When I started to work on this (some time ago) there was a strange atomatic formating from eclipse.
But it looks good now, so I removed the comments (also for the other components)